ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -109,6 +110,111 @@ class Endpoints(
   ): Route =
     responseToRoute(httpResponse(res))
 
+  private def toPostRoute[Req: JsonReader, Res: JsonWriter](
+      httpRequest: HttpRequest,
+      fn: (Jwt, Req) => ET[domain.SyncResponse[Res]],
+  )(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
+      metrics: HttpJsonApiMetrics,
+  ): Route = {
+    val res = for {
+      t <- routeSetup.inputJsVal(httpRequest): ET[(Jwt, JsValue)]
+      (jwt, reqBody) = t
+      req <- either(SprayJson.decode[Req](reqBody).liftErr(InvalidUserInput)): ET[Req]
+      res <- eitherT(RouteSetup.handleFutureEitherFailure(fn(jwt, req).run)): ET[
+        domain.SyncResponse[Res]
+      ]
+    } yield res
+    responseToRoute(httpResponse(res))
+  }
+
+  private def toGetRoute[Res](
+      httpRequest: HttpRequest,
+      fn: (Jwt) => ET[domain.SyncResponse[Res]],
+  )(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
+      mkHttpResponse: MkHttpResponse[ET[domain.SyncResponse[Res]]],
+  ): Route = {
+    val res = for {
+      t <- eitherT(routeSetup.input(httpRequest)): ET[(Jwt, String)]
+      (jwt, _) = t
+      res <- eitherT(RouteSetup.handleFutureEitherFailure(fn(jwt).run)): ET[
+        domain.SyncResponse[Res]
+      ]
+    } yield res
+    responseToRoute(httpResponse(res))
+  }
+
+  private def toGetRouteLedgerId[Res](
+      httpRequest: HttpRequest,
+      fn: (Jwt, LedgerApiDomain.LedgerId) => ET[domain.SyncResponse[Res]],
+  )(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
+      mkHttpResponse: MkHttpResponse[ET[domain.SyncResponse[Res]]],
+  ): Route = {
+    val res = for {
+      t <- extractJwtAndLedgerId(httpRequest)
+      (jwt, ledgerId) = t
+      res <- eitherT(
+        RouteSetup.handleFutureEitherFailure(fn(jwt, ledgerId).run)
+      ): ET[domain.SyncResponse[Res]]
+    } yield res
+    responseToRoute(httpResponse(res))
+  }
+
+  private def toUploadDarFileRoute(httpRequest: HttpRequest)(implicit
+      lc: LoggingContextOf[InstanceUUID with RequestID],
+      metrics: HttpJsonApiMetrics,
+      mkHttpResponse: MkHttpResponse[ET[domain.SyncResponse[Unit]]],
+  ): Route = {
+    val res: ET[domain.SyncResponse[Unit]] = for {
+      parseAndDecodeTimer <- routeSetup.getParseAndDecodeTimerCtx()
+      _ <- EitherT.pure(metrics.uploadPackagesThroughput.mark())
+      t2 <- eitherT(routeSetup.inputSource(httpRequest))
+      (jwt, payload, source) = t2
+      _ <- EitherT.pure(parseAndDecodeTimer.stop())
+
+      _ <- eitherT(
+        RouteSetup.handleFutureFailure(
+          packageManagementService.uploadDarFile(
+            jwt,
+            toLedgerId(payload.ledgerId),
+            source.mapMaterializedValue(_ => NotUsed),
+          )
+        )
+      ): ET[Unit]
+    } yield domain.OkResponse(())
+    responseToRoute(httpResponse(res))
+  }
```

Comment:
It's more important that the route implementation remain separated than that the `routeSetup` argument is removed. So this should move back to where it was for this PR.